scriptmodules/supplementary/skyscraper.sh

```diff
@@ -0,0 +1,696 @@
+#!/usr/bin/env bash -x
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="skyscraper"
+rp_module_desc="Scraper for EmulationStation by Lars Muldjord"
+rp_module_licence="GPLv3.0 https://raw.githubusercontent.com/muldjord/skyscraper/master/LICENSE"
+rp_module_section="opt"
+
+function depends_skyscraper() {
+    getDepends qt5-default p7zip-full
+}
+
+function sources_skyscraper() {
+    local latest_ver
+    latest_ver=$(_latest_ver_skyscraper)
+
+    gitPullOrClone "$md_build" "https://github.com/muldjord/skyscraper" "$latest_ver"
+}
+
+function build_skyscraper() {
+    qmake
+    make
+    md_ret_require="$md_build/Skyscraper"
+}
+
+function install_skyscraper() {
+    md_ret_files=(
+    'Skyscraper'
+    'LICENSE'
+    'README.md'
+    )
+}
+
+# Purge all Skyscraper caches
+function _purge_skyscraper() {
+    local platform
+
+    while read platform; do
+        # Find any folders underneath the Local DB cache and clear them
+        _clear_platform_skyscraper "$platform"
+    done < <(find "$configdir/all/skyscraper/dbs" -maxdepth 1 -mindepth 1 -type d -exec basename {} \;)
+}
+
+function _clear_platform_skyscraper() {
+    local platform="$1"
+
+    [[ ! -d "$configdir/all/skyscraper/dbs/$platform" ]] && return
+
+    # Remove any folder underneath the platform, try to keep any customized files
+    pushd "$configdir/all/skyscraper/dbs/$platform"
+    find . -maxdepth 1 -mindepth 1 -type d -print0 -exec rm -fr {} \;
+    rm -f db.xml
+    popd
+
+}
+# Purge by cached platform
```

Comment:
white space above in function.

no gap between functions.

Reply:
OK, fixed.